lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncTool.java

```diff
@@ -0,0 +1,332 @@
+        configProvider);
+    HoodiePropertiesReader hoodiePropertiesReader = new HoodiePropertiesReader(asyncStorageClient,
+        lakeViewExtractorMetrics);
+    AsyncHttpClientWithRetry asyncHttpClientWithRetry = getAsyncHttpClientWithRetry(executorService);
+    OnehouseApiClient onehouseApiClient = new OnehouseApiClient(asyncHttpClientWithRetry, config,
+        lakeViewExtractorMetrics);
+    PresignedUrlFileUploader presignedUrlFileUploader = new PresignedUrlFileUploader(asyncStorageClient,
+        asyncHttpClientWithRetry, lakeViewExtractorMetrics);
+    TimelineCommitInstantsUploader timelineCommitInstantsUploader = new TimelineCommitInstantsUploader(asyncStorageClient,
+        presignedUrlFileUploader, onehouseApiClient, storageUtils, executorService, new ActiveTimelineInstantBatcher(config),
+        lakeViewExtractorMetrics, config);
+    TableMetadataUploaderService tableMetadataUploaderService = new TableMetadataUploaderService(hoodiePropertiesReader,
+        onehouseApiClient, timelineCommitInstantsUploader, lakeViewExtractorMetrics, executorService);
+    return new TableDiscoveryAndUploadJob(tableDiscoveryService, tableMetadataUploaderService, lakeViewExtractorMetrics);
+  }
+
+  private AsyncStorageClient getAsyncStorageClient(@Nonnull Config config, @Nonnull ExecutorService executorService,
+                                                   StorageUtils storageUtils) {
+    if (config.getFileSystemConfiguration().getS3Config() != null) {
+      S3AsyncClientProvider s3AsyncClientProvider = new S3AsyncClientProvider(config, executorService);
+      return new S3AsyncStorageClient(s3AsyncClientProvider, storageUtils, executorService);
+    } else {
+      GcsClientProvider gcsClientProvider = new GcsClientProvider(config);
+      return new GCSAsyncStorageClient(gcsClientProvider, storageUtils, executorService);
+    }
+  }
+
+  private AsyncHttpClientWithRetry getAsyncHttpClientWithRetry(@Nonnull ExecutorService executorService) {
+    Dispatcher dispatcher = new Dispatcher(executorService);
+    OkHttpClient okHttpClient = new OkHttpClient.Builder()
+        .readTimeout(httpClientTimeoutSeconds, TimeUnit.SECONDS)
+        .writeTimeout(httpClientTimeoutSeconds, TimeUnit.SECONDS)
+        .connectTimeout(httpClientTimeoutSeconds, TimeUnit.SECONDS)
+        .dispatcher(dispatcher)
+        .build();
+    return new AsyncHttpClientWithRetry(
+        httpClientMaxRetries, httpClientRetryDelayMs, okHttpClient);
+  }
+
+  @VisibleForTesting
+  public @Nullable Config getConfig() {
+    return config;
+  }
+
+  @Override
+  public void syncHoodieTable() {
+    if (isLakeviewSyncToolEnabled && tableDiscoveryAndUploadJob != null) {
+      Future<?> future = executorService.submit(tableDiscoveryAndUploadJob::runOnce);
+      try {
+        if (timeoutInSeconds > 0) {
+          future.get(timeoutInSeconds, TimeUnit.SECONDS);
+        } else {
+          future.get();
```

Comment:
should we consider having a default timeout? first time ingestion can take hours sometimes

Reply:
we provide a timeout to be configured by users if needed. i'll check with Andy on the default timeout of this separately.